Monika After Story/game/zz_submods.rpy

```diff
@@ -0,0 +1,318 @@
+                        ).lower()
+                    )
+
+                #Even if this hasn't updated, we should adjust its value to reflect the correct version
+                persistent._mas_submod_version_data[submod.name] = submod.version
+
+        @staticmethod
+        def _checkDependencies():
+            """
+            Checks to see if the dependencies for this submod are met
+            """
+            for submod in submod_map.itervalues():
+                for dependency, minmax_version_tuple in submod.dependencies.iteritems():
+                    dependency_submod = Submod._getSubmod(dependency)
+
+                    if dependency_submod is not None:
+                        #Now we need to split our minmax
+                        minimum_version, maximum_version = minmax_version_tuple
+
+                        #First, check the minimum version
+                        if (
+                            minimum_version
+                            and not Submod._checkVersions(dependency_submod.version, minimum_version)
+                        ):
+                            raise SubmodError(
+                                "Submod '{0}' is out of date. Version {1} required.".format(
+                                    dependency_submod.name, minimum_version
+                                )
+                            )
+
+                        #If we have a maximum version, we should check if we're surpassing it
+                        elif (
+                            maximum_version
+                            and Submod._checkVersions(dependency_submod.version, maximum_version, is_current=False)
+                        ):
+                            raise SubmodError(
+                                "Version '{0}' of '{1}' is installed and is incompatible with {2}.\nVersion {3} is compatible.".format(
+                                    dependency_submod.version, dependency_submod.name, submod.name, maximum_version
+                                )
+                            )
+
+                    #Submod wasn't installed at all
+                    else:
+                        raise SubmodError(
+                            "Submod '{0}' is not installed and is required for {1}.".format(
+                                dependency, submod.name
+                            )
+                        )
+
+        @staticmethod
+        def _getSubmod(name):
+            """
+            Gets the submod with the name provided
+
+            IN:
+                name - name of the submod to get
+
+            OUT:
+                Submod object representing the submod by name if installed and registered
+                None if not found
+            """
+            return submod_map.get(name)
+
+        @staticmethod
+        def _checkVersions(curr_vers, comparitive_vers, is_current=True):
```

Comment:
so a comparison function like this where we want to know if something is greater/less/equal, is normally created as a `compareTo` method that returns 1,0, or -1 if the left arg is greater than, equal, or less than the right arg. I would restructure this function to be a `compareTo` style function so `is_current` is not needed.

Also I would make this **not** a static method (and have the "left arg" be considered itself) since all the places that this function is used refers to a submod object anyway. If you forsee the using the version checker for non submod-related version checking, then make the meat of the version check in `mas_utils` and have a class function here that calls the meat.

Reply:
Will do.